tests/semantic_analyser.cpp

```diff
@@ -3012,96 +3198,144 @@ class semantic_analyser_btf : public test_btf {};
-  test("kretfunc:func_1 { $k = path( retval->foo1 ) }", 0);
+  test("kfunc:func_1 { $k = path( args.foo1 ) }");
+  test("kretfunc:func_1 { $k = path( retval->foo1 ) }");
 }
 
 TEST_F(semantic_analyser_btf, call_skb_output)
 {
-  test("kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1, 1500, 0); }",
-       0);
+  test("kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1, 1500, 0); }");
   test("kretfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1, 1500, 0); "
-       "}",
-       0);
+       "}");
 
-  test("kfunc:func_1 { $ret = skboutput(); }", 1);
-  test("kfunc:func_1 { $ret = skboutput(\"one.pcap\"); }", 1);
-  test("kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1); }", 1);
-  test("kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1, 1500); }", 1);
-  test("kfunc:func_1 { skboutput(\"one.pcap\", args.foo1, 1500, 0); }", 1);
+  test_error("kfunc:func_1 { $ret = skboutput(); }", R"(
+stdin:1:23-34: ERROR: skboutput() requires 4 arguments (0 provided)
+kfunc:func_1 { $ret = skboutput(); }
+                      ~~~~~~~~~~~
+)");
+  test_error("kfunc:func_1 { $ret = skboutput(\"one.pcap\"); }", R"(
+stdin:1:23-44: ERROR: skboutput() requires 4 arguments (1 provided)
+kfunc:func_1 { $ret = skboutput("one.pcap"); }
+                      ~~~~~~~~~~~~~~~~~~~~~
+)");
+  test_error("kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1); }", R"(
+stdin:1:23-55: ERROR: skboutput() requires 4 arguments (2 provided)
+kfunc:func_1 { $ret = skboutput("one.pcap", args.foo1); }
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+)");
+  test_error(
+      "kfunc:func_1 { $ret = skboutput(\"one.pcap\", args.foo1, 1500); }", R"(
+stdin:1:23-61: ERROR: skboutput() requires 4 arguments (3 provided)
+kfunc:func_1 { $ret = skboutput("one.pcap", args.foo1, 1500); }
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+)");
+  test_error("kfunc:func_1 { skboutput(\"one.pcap\", args.foo1, 1500, 0); }",
+             R"(
+stdin:1:16-57: ERROR: skboutput() should be assigned to a variable
+kfunc:func_1 { skboutput("one.pcap", args.foo1, 1500, 0); }
+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+)");
 }
 
 TEST_F(semantic_analyser_btf, iter)
 {
-  test("iter:task { 1 }", 0);
-  test("iter:task_file { 1 }", 0);
-  test("iter:task_vma { 1 }", 0);
-  test("iter:task { $x = ctx->task->pid }", 0);
-  test("iter:task_file { $x = ctx->file->ino }", 0);
-  test("iter:task_vma { $x = ctx->vma->vm_start }", 0);
-  test("iter:task { $x = args.foo; }", 1);
-  test("iter:task_file { $x = args.foo; }", 1);
-  test("iter:task_vma { $x = args.foo; }", 1);
-  test("iter:task { printf(\"%d\", ctx->task->pid); }", 0);
-  test("iter:task_file { printf(\"%d\", ctx->file->ino); }", 0);
-  test("iter:task_vma { printf(\"%lx\", ctx->vma->vm_start); }", 0);
-  test("iter:task,iter:task_file { 1 }", 1);
-  test("iter:task,iter:task_vma { 1 }", 1);
-  test("iter:task,f:func_1 { 1 }", 1);
+  test("iter:task { 1 }");
```

Comment:
A few of these test cases were just checking the exact same thing for each attachment point, so weren't giving us any extra test coverage. Deleted them to make my life easier